db/migrate/20191213123801_create_whitehall_imported_assets.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class CreateWhitehallImportedAssets < ActiveRecord::Migration[6.0]
+  def change
+    create_table :whitehall_imported_assets do |t|
+      t.references :whitehall_import,
+                   foreign_key: { to_table: :whitehall_imports,
+                                  on_delete: :restrict },
+                   index: true,
+                   null: false
+
+      # Asset must be one of FileAttachment or Image
+      # @TODO - write a custom validator?
+      t.references :file_attachment_revision,
+                   foreign_key: { to_table: :file_attachment_revisions,
+                                  on_delete: :cascade }
+      t.references :image_revision,
+                   foreign_key: { to_table: :image_revisions,
+                                  on_delete: :cascade }
+
+      t.string :original_asset_url, null: false
+      t.string :variants, json: true, default: "{}"
```

Comment:
I think rather than dealing with a collection of asset url and variants it might make sense for each record in this to be an individual asset_url. The thing I'm thinking is if there are multiple assets on a single record then a single state doesn't really make sense as one asset can be successful and another one fail for instance.

I anticipated it to be that we'd have a table like:

asset_url
image_revision
file_attachment_revision
image_revision_variant
state

and then we could have a destination_url method on the model like so:

```
def destination_url
  return file_attachment_revision.asset_url if file_attachment_revision
  return image_revision.asset_url(image_revision_variant) if image_revision
end
```

then can use a string from this to redirect and a nil from it to do a removal.

Reply:
Thanks for the idea. I like it.

It took quite a bit of reworking so I've created a new PR: https://github.com/alphagov/content-publisher/pull/1568

I'll close this PR but won't delete the branch, that way I can roll back to this version if needed.